float64[] action
int64 act_len
---
float64[] par_upd
int64 par_len
float64 obs
float64 coverage
float64 path_len